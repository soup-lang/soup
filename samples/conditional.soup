func aAnd(x, y, %%)
    setTimeout(%, 1)
    if !x
        setTimeout(%, 1)
        return false
    setTimeout(%, 1)
    return y

func aNeg(x, %%)
    setTimeout(%, 1)
    return !x

func sAnd(x, y)
    return x && y

func sNeg(x)
    return !x

console.log('Group A')
console.log('00', aNeg(false, %%) if aAnd(true, true, %%) else sAnd(true, false))
console.log('01', sNeg(false) if aAnd(true, true, %%) else aAnd(true, false, %%))
console.log('02', aNeg(false, %%) if aAnd(true, true, %%) else aAnd(true, false, %%))
console.log('03', sNeg(false) if aAnd(true, true, %%) else sAnd(true, false))
console.log('04', sNeg(false) if sAnd(true, true) else sAnd(true, false))
console.log('Group B')
console.log('10', aAnd(true, true, %%) if aNeg(true, %%) else sAnd(true, false))
console.log('11', aAnd(true, true, %%) if sNeg(true) else aAnd(true, false, %%))
console.log('12', aAnd(true, true, %%) if aNeg(true, %%) else aAnd(true, false, %%))
console.log('13', aAnd(true, true, %%) if sNeg(true) else sAnd(true, false))
console.log('14', sAnd(true, true) if sNeg(true) else sAnd(true, false))
console.log('End')
