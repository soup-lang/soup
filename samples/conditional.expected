Group A
00 true
01 true
02 true
03 true
04 true
Group B
10 false
11 false
12 false
13 false
14 false
End
